valohai_yaml/objs/pipelines/execution_node.py

```diff
@@ -3,42 +3,55 @@
 from valohai_yaml.lint import LintResult
 from valohai_yaml.objs.pipelines.node import ErrorAction, Node
 from valohai_yaml.objs.pipelines.node_action import NodeAction
-from valohai_yaml.types import LintContext, NodeOverrideDict
+from valohai_yaml.objs.pipelines.override import Override
+from valohai_yaml.types import LintContext, SerializedDict
 
 
 class ExecutionNode(Node):
     """Represents an execution node within a pipeline definition."""
 
     type = 'execution'
-
+    override: Override
```

Comment:
`override` is probably the less common case, so it might be better to just not have one if it would be empty?
```suggestion
    override: Optional[Override]
```
(Also requires the relevant change in the constructor to make `.override` None if an empty dict is passed in.)

Reply:
if it is optional, it would lean to overcomplexity in lint()

Reply:
You can refactor `lint()` to call out to a separate `_lint_override()` if an override is present.